### DISCUSSION TITLE
How to connect with AWS IOT MQTT core with only having brokerurl and ClientId

### DISCUSSION THREAD

**Discussion by harialgat** [aws/aws-iot-device-sdk-java-v2#388] [Q&A] [answered]
I am testing MQTT protocol I don't have any credentials I only have to broker url and clientId. How do I connect to it using java code?

**Comment by MikeDombo** [comment]
You cannot connect to AWS IoT without credentials.

https://docs.aws.amazon.com/iot/latest/developerguide/iot-connect-devices.html

**Comment by bretambrose** [comment]
Mike is correct with respect to AWS IoT Core.  However if you wish to test against a local broker that does not require authentication, you can do so by creating the client directly rather than using the builder class (which does lots of IoT Core specific set up for you).  

If you're using MQTT5 you can use the [CRT MQTT5 tests ](https://github.com/awslabs/aws-crt-java/blob/main/src/test/java/software/amazon/awssdk/crt/test/Mqtt5ClientTest.java)as a model (most of the parameters set there are not needed for basic testing purposes).

If you're using MQTT311 you can use the [raw connect sample](https://github.com/aws/aws-iot-device-sdk-java-v2/blob/main/samples/RawConnect/src/main/java/rawconnect/RawConnect.java#L112-L120) as a starting point (however, you probably wouldn't be using TLS and so you'd use the MqttClient constructor variant that takes nothing).

**Reply by harialgat** [reply]
```
 public void initiateMQTTConnection(MqttParams params) {
        MqttClient client = new MqttClient();
        MqttConnectionConfig config = new MqttConnectionConfig();
        config.setMqttClient(client);
        config.setClientId(params.clientId);
        config.setEndpoint(params.url);
        config.setCleanSession(true);
        config.setConnectionCallbacks(new MqttClientConnectionEvents() {
            @Override
            public void onConnectionInterrupted(int errorCode) {

            }

            @Override
            public void onConnectionResumed(boolean sessionPresent) {

            }
        });
        MqttClientConnection connection = new MqttClientConnection(config);
        connection.connect();

    }
```

after using this code getting the exception: Exception in thread "main" software.amazon.awssdk.crt.mqtt.MqttException: port must be a positive integer between 1 and 65535 

The url I am using does not have the port in it and it starts with wss

**Reply by jmklix** [reply, marked as answer]
Can you try port 443?

As noted on [this](https://datatracker.ietf.org/doc/id/draft-ietf-hybi-thewebsocketprotocol-09.html#:~:text=WebSocket%20URIs,-This%20specification%20defines&text=The%20port%20component%20is%20OPTIONAL,%22wss%22%20is%20port%20443.) webpage the default for wss is 443
> The port component is OPTIONAL; the default for "ws" is port 80, while the default for "wss" is port 443.
